src/robothub_oak/utils.py

```diff
@@ -1,57 +1,93 @@
-import logging
+import logging as log
+from typing import Dict, Any, Union
 
-from typing import Dict, Any
+import depthai
+import robothub_core
 
-__all__ = [
-    'try_or_default',
-    'set_logging_level',
-    '_process_kwargs',
-    '_get_methods_by_class',
-    '_convert_to_enum'
-]
 
+def set_logging_level(level: Union[str, int]) -> None:
+    """
+    Set the logging level for the application.
 
-def try_or_default(func, default=None):
-    try:
-        return func()
-    except:
-        return default
+    :param level: Either a string or an integer. If a string, it must be one of the following: 'DEBUG', 'INFO',
+     'WARNING', 'ERROR', 'CRITICAL'. If an integer, it must be one of the following: log.DEBUG, log.INFO, log.WARNING,
+      log.ERROR, log.CRITICAL.
+    """
+    if isinstance(level, str):
+        level = level.upper()
 
+    if level != 'DEBUG' or level != log.DEBUG:
+        log.basicConfig(format='%(levelname)s | %(message)s')
+    else:
+        log.basicConfig()
 
-def set_logging_level(level):
-    logging.basicConfig()
-    logging.getLogger().setLevel(level)
+    log.getLogger().setLevel(level)
 
 
-def _process_kwargs(kwargs: Dict[str, Any]) -> Dict[str, Any]:
-    """Process the kwargs and remove all None values."""
-    kwargs.pop('self')
-    kwargs = {k: v for k, v in kwargs.items() if v is not None}
-    return kwargs
+def stats_report(device: depthai.Device) -> Dict[str, Any]:
+    """
+    Returns a dictionary with statistics about the device.
+    """
+    stats = {'mxid': device.getMxId()}
 
+    css_cpu_usage = device.getLeonCssCpuUsage().average
+    mss_cpu_usage = device.getLeonMssCpuUsage().average
+    cmx_mem_usage = device.getCmxMemoryUsage()
+    ddr_mem_usage = device.getDdrMemoryUsage()
+    chip_temp = device.getChipTemperature()
 
-def _get_methods_by_class(cls: Any) -> list:
-    """
-    Get all methods of a class that are not private.
+    stats['css_usage'] = int(100 * css_cpu_usage)
+    stats['mss_usage'] = int(100 * mss_cpu_usage)
+    stats['ddr_mem_free'] = int(ddr_mem_usage.total - ddr_mem_usage.used)
+    stats['ddr_mem_total'] = int(ddr_mem_usage.total)
+    stats['cmx_mem_free'] = int(cmx_mem_usage.total - cmx_mem_usage.used)
+    stats['cmx_mem_total'] = int(cmx_mem_usage.total)
+    stats['css_temp'] = int(100 * chip_temp.css)
+    stats['mss_temp'] = int(100 * chip_temp.mss)
+    stats['upa_temp'] = int(100 * chip_temp.upa)
+    stats['dss_temp'] = int(100 * chip_temp.dss)
+    stats['temp'] = int(100 * chip_temp.average)
 
-    :param cls: The class to get the methods from.
-    :return: A list of methods.
-    """
-    return [x for x in dir(cls) if callable(getattr(cls, x)) and not x.startswith('_')]
+    return stats
 
 
-def _convert_to_enum(name: Any, cls: Any) -> Any:
+def info_report(device: depthai.Device, state: robothub_core.DeviceState) -> Dict[str, Any]:
```

Comment:
I think we cal think of a better name here as well. My suggestions:

- get_device_attributes
- get_device_specifications
- get_device_details